co-log-core/src/Colog/Core/Class.hs

```diff
@@ -18,11 +17,12 @@ TODO: laws
 class HasLog env msg m where
     getLogAction :: env -> LogAction m msg
     setLogAction :: LogAction m msg -> env -> env
+    setLogAction = seq
+    overLogAction :: (LogAction m msg -> LogAction m msg) -> env -> env
+    overLogAction = seq
```

Comment:
Also, not `seq`, you should use `getLogAction` and `setLogAction` to implement `overLogAction`